File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {

        set<char> val;
        int maxLen=0;
        int left=0;
        for(int i=0;i<s.length();i++){
            if(val.count(s[i])==0){
                val.insert(s[i]);
               maxLen=max(maxLen,i-left+1);
            }
            else{
                 while (val.count(s[i])==1){
                    val.erase(s[left]);
                    left++;
            }
              val.insert(s[i]);
            }
           
        }
        return maxLen;
        
    }
};

File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        max_len=0
        temp=list()
        for i in s:
            if i not in temp:
                temp.append(i)
            else:
                max_len=max(max_len,len(temp))
                temp = temp[temp.index(i) + 1:]
                temp.append(i)
        max_len = max(max_len, len(temp)) 
        return max_len